ext/OceananigansAMDGPUExt/DistributedComputations.jl

```diff
@@ -0,0 +1,25 @@
+module DistributedComputations
+
+using ..Architectures: ROCArray, ROCmGPU
+
+using Oceananigans.DistributedComputations: DistributedField
+
+import Oceananigans.DistributedComputations: set!
+
+function set!(u::DistributedField, v::ROCArray)
+    gsize = global_size(architecture(u), size(u))
+
+    if size(v) == size(u)
+        f = arch_array(architecture(u), v)
+        u .= f
+        return u
+    elseif size(v) == gsize
+        f = partition_global_array(architecture(u), v, size(u))
+        u .= f
+        return u
+    else
+        throw(ArgumentError("ERROR: DimensionMismatch: array could not be set to match destination field"))
+    end
+end
```

Comment:
Not sure that we need this specialization here. This `set!` should be architecture agnostic. You can just modify the original `set!` to have `v::AbstractArray` as an argument

Reply:
@simone-silvestri means to change this line:

https://github.com/CliMA/Oceananigans.jl/blob/f2511962ca15f3aaf87d2571e3551e59dc05c694/src/DistributedComputations/distributed_fields.jl#L39

to dispatch on `AbstractArray` rather than `Union{Array, CuArray}`.

Reply:
Took care of this with https://github.com/fluidnumerics-joe/Oceananigans.jl/commit/455262253154cef892c61048ceb3eca500a23494

Tested it out on our systems and all is good to go here.